### PR TITLE
Remove dependency on TimescaleDB image and support base backups via HTTP API

### PR DESCRIPTION
**Pull request by erimatnor**
The WAL-E image previously had a dependency on the TimescaleDB container, which had to be customized to work with WAL-E. Further, the containers were setup to synchronize startup via a lock file to support restore from a base backup. This was hard to configure and was error prone.

The WAL-E image is now greatly simplified, essentially used bundling WAL-E with a HTTP API. The API has been expanded to allow triggering base backups via HTTP.

Restore is no longer handled explicitly by the image. This is better handled by the user for their particular setup. For instance, base backup restores are easily implemented in Kubernetes init containers that run before the main Postgres/TimescaleDB container.

### CODE REVIEW

**Comment by doomsower** [comment]
Any chance to see this merged? Also, while it's not merged and pushed to registry, do you think it's safe to use it directly from your fork?

File: README.md [hunk 1]
```diff
@@ -1,64 +1,71 @@
...
+`PGDATA` and `PGWAL` disk volumes so that the WAL-E sidecar container
+can access the database files it needs to backup.
+
+### TimescaleDB / PostgreSQL configuration
+To enable continuous archiving for the PostgreSQL WAL, the following
+modifications are necessary in `postgresql.conf`:
+
+```
+archive_command='wget <wale_container_hostname>/wal-push/%f -O -'
 wal_level=archive
```
Comment (davidkohn88) [inline README.md]: wal_level=archive is supported, but mapped to replica as of PG 9.6, so we should probably just do wal_level=replica

**Review by RobAtticus — APPROVED** [review]

**Comment by RobAtticus** [comment]
Thanks for this @erimatnor , we'll also need to update the docs.

@doomsower - Sorry this slipped through the cracks with some other stuff going on, but it should get merged soon.

**Review by davidkohn88 — COMMENTED** [review]
Sorry for late review here, one nit to pick on wal_level, but otherwise, looks good to me.

File: README.md (hunk 1 above)
Comment (erimatnor) [reply README.md]: Cool, fixed.

**Comment by erimatnor** [comment]
@RobAtticus @davidkohn88 Btw, note that I set the default image name to `timescale/wal-e`  in the Makefile. We've pushed this image to `timescale/timescaledb-wale` before but I think the new default image name makes more sense since there is no longer a dependency in TimescaleDB (i.e., this image is mostly just dockerized WAL-E). Also, I think it would be awkward to prefix everything with `timescaledb` for no obvious reason when we are already namespacing with `timescale/`. 

We can still push to the old name for compatibility or we can just deprecate it. We might also consider renaming this repo. I don't think it should be a major problem since I don't think this is a widely used repo, at least not yet.

Thoughts?

**Comment by RobAtticus** [comment]
I am fine with renaming the docker image to what you suggested. Not sure if renaming this repo is a good idea since it could be confused as our fork of wal-e rather than a dockerized wal-e that is useful for timescaledb.

**Review by LeeHampton — APPROVED** [review]

**Pull request merged**